lib/Runtime/Language/JavascriptConversion.cpp

```diff
@@ -701,8 +701,16 @@ using namespace Js;
         case TypeIds_Number:
             return JavascriptNumber::ToLocaleString(JavascriptNumber::GetValue(aValue), scriptContext);
 
-        case TypeIds_String:
-            return UnsafeVarTo<JavascriptString>(aValue);
+        case TypeIds_String: {
+            RecyclableObject* object = VarTo<RecyclableObject>(aValue);
+            Var value = JavascriptOperators::GetProperty(object, PropertyIds::toLocaleString, scriptContext, NULL);
+            RecyclableObject* toLocaleStringFunction = VarTo<RecyclableObject>(value);
```

Comment:
`VarTo<RecyclableObject>(value)` will crash the program if `value` is not a recyclable object (suppose maybe it's an int, for example).